taichi/transforms/alg_simp.cpp

```diff
@@ -100,6 +100,11 @@ class AlgSimp : public BasicStmtVisitor {
 
   bool optimize_multiplication(BinaryOpStmt *stmt) {
     // return true iff the IR is modified
+    if (stmt->lhs->ret_type->is<TensorType>() ||
```

Comment:
I guess this check is already covered by the one in `void visit(BinaryOpStmt *stmt)`.